File: ArraySort/python/insertionSort.py
def insertionSort(array):
    for i in range(1, len(array)):
        j = i
        print("for Loop I index", i)
        while j > 0 and array[j] < array[j - 1]:
            print("Before J Index ", j)
            swap(j, j - 1, array)
            j -= 1
            print("After J Index", j)
            print(array)
    return array


def swap(i, j, array):
    array[i], array[j] = array[j], array[i]


arr = [22, 3, 2, 1, 41]
print(insertionSort(arr))

File: ArraySort/js/insertionSort.js
const swap = (a, b, arr) => {
  [arr[a], arr[b]] = [arr[b], arr[a]];
};

const insertionSort = (arr) => {
  for (let i = 1; i < arr.length; i++) {
    let j = i;
    while (j > 0 && arr[j] < arr[j - 1]) {
      swap(j, j - 1, arr);
      j--;
    }
  }
  return arr;
};

const arr = [4, 1, 8, 2, 3, 2];
console.log(insertionSort(arr));
